### DISCUSSION TITLE
Questions Regarding upcoming Load Balancing Feature in Umbraco Cloud

### DISCUSSION THREAD

**Discussion by Nikhilgirirajdigital** [umbraco/Umbraco.Cloud.Issues#896] [Ideas]
### Issue description

At Codegarden, we were thrilled to hear about the upcoming load-balancing feature in Umbraco 
Cloud. As we currently host load-balanced Umbraco websites on Azure using Umbraco CMS [as per 
the documentation](https://docs.umbraco.com/umbraco-cms/fundamentals/setup/server-setup/load-balancing/azure-web-apps), we are eager to understand how this new feature will impact our existing setups 
and future migration plans. We have prepared a series of questions to gain clarity on this matter, and 
we would be grateful if you could help us with the following inquiries:

1. Will the introduction of load balancing in Umbraco Cloud allow us to migrate our load-balanced 
websites, currently hosted on Azure using Umbraco CMS, to the Umbraco Cloud platform? If yes, 
what steps or considerations should we be aware of during the migration process?

2. Considering that the publisher instance can only be on a single instance, will it be possible to host 
the publisher on shared hosting while having the subscriber on a dedicated server? We came across 
a related issue (#613 on GitHub), but it does not seem to specifically address load balancing.

3. Will the subscriber instances automatically scale based on resource utilization and traffic, or will 
Do we need to scale them out manually through the portal?

4. With regards to scaling subscriber instances, will they be distributed across various regions 
supported in Umbraco Cloud, or will they be limited to a specific region? Additionally, since Cloud 
currently does not support a dedicated SQL elastic pool (#658 on GitHub), will load balancing also 
utilize shared SQL Compute with other Cloud customers?

5. Could you provide an overview of the pricing model for setting up a load-balanced environment on 
Umbraco Cloud, considering the answers to the previous questions? This will help us assess the 
feasibility and cost implications of migrating from Azure to Umbraco Cloud.

6. Will it be possible to have a staging environment that is load balanced for conducting load testing? 
Or will this feature only be available for live environments?

7. In the documentation, there is a warning about not editing templates or views on a live server due 
to discrepancies between SchedulingPublisher and Subscriber environments' file systems. Is there a 
way to disable the capability of making changes in the Live back office to prevent accidental 
disruptions?

8. Will the feature work for all versions of Umbraco? Including v7?

9. Lastly, to ensure we have a comprehensive understanding of the load balancing feature and its 
implications for our specific use case, would it be recommended to book an Architecture Advising 
session with the Umbraco Cloud team? If yes, how can we proceed with scheduling such a session?

Thank you in advance for your time and consideration in addressing these questions. We eagerly 
anticipate the arrival of the load balancing feature in Umbraco Cloud and look forward to making the 
most of its benefits for our projects

**Comment by sajumb** [comment]
Thank you for your enthusiasm about the upcoming load-balancing feature in Umbraco Cloud! While we're excited to introduce this feature, we have not yet outlined all the details. We appreciate your comprehensive set of questions, which shows your dedication to understanding how this will impact your existing setups and future plans. At this time, we're actively working on finalizing the specifics for some of the enterprise features that will be launched prior to the load-balancing feature.

Given the complexity of your inquiries, we cannot answer many of them in great detail. Here are some of the information that we do know at the moment:

**Best Practices for Load Balancing:**
We expect to follow the best practices regarding load balancing and in especially follow the methods and content currently contained in the "load balancing" course from Umbraco Training. Our goal is to ensure a smooth and effective transition for your load-balanced setups, aligning with the practices that have proven successful.

**Publisher and Subscriber Relationship:**
We have currently not defined the specifics of the publisher and subscriber relationship within the load-balancing setup. Our team is meticulously evaluating different approaches to provide you with the best possible experience.

**Pricing Considerations:**
At this stage, we've not yet formulated the pricing model for the load-balancing feature. Rest assured, our aim is to provide a pricing structure that reflects the value and capabilities this feature will bring to your projects.

**Compatibility with Umbraco Versions:**
We have not decided whether Umbraco 7 will be supported in the upcoming load-balancing feature. My current guess will be that it won't be supported. Umbraco 7 has reached its end-of-life support and Umbraco 7's extended security support period is concluding soon.

**Architecture Advising Session Timing:**
We appreciate your interest in our Architecture Advising sessions. While you're certainly welcome to schedule a session, we recommend waiting a bit until we can offer a comprehensive overview of the feature set. This will ensure that your advisory session is as informative and tailored to your needs as possible.

We understand that you're eager for more details, and we'll certainly keep you updated as we progress with the development and documentation of the load-balancing feature. Your feedback and questions are valuable to us as we work towards making this feature a valuable addition to your Umbraco experience. Thank you for your patience and anticipation!

**Comment by stephengeospy** [comment]
Hey Cloud Team,

Just had a quick question pop up. With the upcoming load balancing feature, will we have the ability to load balance the Umbraco backoffice server as well? You know, to manage a large number of editors all working at the same time. Or is it going to be a single instance deal even after the feature rolls out?

Thanks!

**Comment by meyntony** [comment]
> Hey Cloud Team,
> 
> Just had a quick question pop up. With the upcoming load balancing feature, will we have the ability to load balance the Umbraco backoffice server as well? You know, to manage a large number of editors all working at the same time. Or is it going to be a single instance deal even after the feature rolls out?
> 
> Thanks!

Thanks for reaching out. Umbraco does not support the scale out of the Server that is serving the Umbraco Backoffice. This is explained in detail in the CMS [docs](https://docs.umbraco.com/umbraco-cms/fundamentals/setup/server-setup/load-balancing/azure-web-apps#scaling)
This means that this instance (the publisher) cannot be scaled out / load balanced, but you can scale it up, In Umbraco Cloud this can be done by using the [Dedicated Resources add on](https://umbraco.com/products/umbraco-cloud/dedicated-resources/).

Only the subscriber instances (Front-end servers) can be scaled out i.e. [load balanced](https://docs.umbraco.com/umbraco-cms/fundamentals/setup/server-setup/load-balancing) 
To learn more, I'd recommend [the Umbraco Training ](https://umbraco.com/training/course-details/load-balancing-and-azure/) as @sajumb mentioned, we expect to follow this when the feature is available in Umbraco Cloud.

> **Best Practices for Load Balancing:**
> We expect to follow the best practices regarding load balancing and in especially follow the methods and content currently contained in the "load balancing" course from Umbraco Training. Our goal is to ensure a smooth and effective transition for your load-balanced setups, aligning with the practices that have proven successful.

**Comment by Nikhilgirirajdigital** [comment]
Will Redis cache be provided for frontend servers i.e. subscribers?

**Comment by ksuvec** [comment]
We do not know that much about the implementation details on the Load Balancing Feature yet.
